src/Microsoft.Data.SqlClient/netfx/src/Microsoft/Data/SqlClient/TdsParser.cs

```diff
@@ -1209,11 +1225,12 @@ private void SendPreLoginHandshake(byte[] instanceName, bool encrypt, bool clien
                                     // Encrypt all.
                                     _encryptionOption = EncryptionOptions.ON | (_encryptionOption & ~EncryptionOptions.OPTIONS_MASK);
                                 }
-
+                                // NOT_SUP: No encryption.
                                 break;
 
                             case (EncryptionOptions.NOT_SUP):
-                                if ((serverOption & EncryptionOptions.OPTIONS_MASK) == EncryptionOptions.REQ)
+                                // Server required encryption
+                                if ((serverOption & EncryptionOptions.OPTIONS_MASK) != EncryptionOptions.NOT_SUP)
```

Comment:
This is technically same as `if ((serverOption & EncryptionOptions.OPTIONS_MASK) == EncryptionOptions.REQ)` as these are the only 2 options supported by server when client sends `EncryptionOptions.NOT_SUP`

![image](https://user-images.githubusercontent.com/13396919/129945906-b42ef49a-d005-4248-a256-27445a476177.png)

Reply:
Correct, just my point was it could be more descriptive, which means we expect to see `NOT_SUP` here. If you still prefer the old condition I can revert it.
(Same explanation for the similar comments.)

Reply:
I would go by old, as it clearly defines "when" to terminate connection.